pathway-analysis/qusage_replicate_vignette.Rmd

```diff
@@ -0,0 +1,404 @@
+  # the subject & timepoint info are combined in the 'title' column
+  dplyr::mutate(subject = stringr::word(title, 1, sep = ","),
+                # relabel the timepoints, the lack of spaces will help
+                # with the contrast call later
+                timepoint = dplyr::case_when(
+                  time_hpi == "Hour 00" ~ "t0",
+                  time_hpi == "Hour 077" ~ "t1"
+                )) %>% 
+  dplyr::arrange(time_hpi, subject) %>%
+  dplyr::select(refinebio_accession_code, subject, refinebio_title, class_blu,
+                clinic_pheno, time_hpi, timepoint)
+```
+
+```{r}
+time_filtered_exprs <- exprs_df %>%
+  dplyr::select(-ENSEMBL) %>%
+  # we're reordering samples here to match the metadata ordering
+  dplyr::select(ENTREZID, filtered_metadata_df$refinebio_accession_code)
+```
+
+Are there duplicated identifiers?
+
+```{r}
+anyDuplicated(time_filtered_exprs$ENTREZID)
+```
+
+Collapse to mean values
+
+```{r}
+collapsed_exprs <- time_filtered_exprs %>%
+  dplyr::group_by(ENTREZID) %>%
+  dplyr::summarise_all(mean) %>%
+  # drop unannotated gene
+  dplyr::filter(!is.na(ENTREZID))
+```
+
+We need to pass `qusage` a matrix.
+Note that we're departing a bit from the advice of the authors by
+refraining from filtering the expression matrix.
+
+```{r}
+exprs_mat <- collapsed_exprs %>%
+  tibble::column_to_rownames("ENTREZID") %>%
+  as.matrix()
+```
+
+Let's get rid of some of the objects in the workspace we don't need.
+
+```{r}
+rm(collapsed_exprs, time_filtered_exprs, exprs_df, metadata_df)
+```
+
+### Read in gene sets
+
+```{r}
+kegg_genesets <- read.gmt(kegg_file)
+```
+
+### Perform pathway analysis
+
+```{r}
+timepoint_qusage_results <- qusage(eset = exprs_mat,
+                                   labels = filtered_metadata_df$timepoint,
+                                   contrast = "t1-t0",
+                                   geneSets = kegg_genesets)
+```
+
+We can look at the results with the `qsTable` function.
+This will give us the top 20 pathways by default.
+
+```{r}
+qsTable(timepoint_qusage_results)
+```
+
+Let's save the results to file.
+First, let's save a nicely formatted table.
+
+```{r}
+timepoint_table <- qsTable(timepoint_qusage_results,
+                           # numPathways let's us snag all tested gene sets
+                           number = numPathways(timepoint_qusage_results))
+readr::write_tsv(timepoint_table,
+                 file.path("results", "qusage_t1-t0_kegg_results.tsv"))
+```
+
+Note that gene sets with negative fold changes are higher in `t0`.
+
+We're also going to save the `QSarray` output of `qusage` because, as we'll see
+shortly, there's lots you can do with qusage results!
+
+```{r}
+readr::write_rds(timepoint_qusage_results,
+                 file.path("results", "qusage_t1-t0_kegg_QSarray.RDS"))
+```
+
+#### Plotting
+
+We can look at the overall results with the `plotCIs` function.
+
+```{r}
+plotCIs(timepoint_qusage_results)
```

Comment:
The x axis label text is so smooshy and hard to read for all their plots. It's a bit better in the pdfs but still not great. You could use `cex.xaxis` argument to try to adjust that. Maybe it would help.